src/event.js

```diff
@@ -363,8 +352,6 @@ jQuery.event = {
 			// Support: Firefox <=42 - 66+
 			// Suppress spec-violating clicks indicating a non-primary pointer button (trac-3861)
 			// https://www.w3.org/TR/DOM-Level-3-Events/#event-type-click
-			// Support: IE 11+
-			// ...but not arrow key "clicks" of radio inputs, which can have `button` -1 (gh-2343)
```

Comment:
Firefox doesn't have the behavior from the above comment (`click` events are not fired for non-primary mouse buttons) so perhaps this all could be removed. But one of our tests checks this condition explicitly so I'd rather leave it for a separate PR.